InventoryInStorePickupShippingApi/Model/IsInStorePickupDeliveryAvailableForCart.php

```diff
@@ -0,0 +1,123 @@
+<?php
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickupShippingApi\Model;
+
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Quote\Api\CartRepositoryInterface;
+use Magento\Quote\Api\Data\CartInterface;
+use Magento\Quote\Api\Data\EstimateAddressInterface;
+use Magento\Quote\Api\Data\EstimateAddressInterfaceFactory;
+use Magento\Quote\Api\Data\ShippingAssignmentInterface;
+use Magento\Quote\Api\ShippingMethodManagementInterface;
+
+/**
+ * Check if In-Store Pickup delivery method is applicable for a cart by cartId.
+ *
+ * @api
+ */
+class IsInStorePickupDeliveryAvailableForCart
+{
+    private const CARRIER_CODE = 'in_store';
+    private const XML_PATH_DEFAULT_COUNTRY = 'general/country/default';
+
+    /**
+     * @var CartRepositoryInterface
+     */
+    private $cartRepository;
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+
+    /**
+     * @var ShippingMethodManagementInterface
+     */
+    private $shippingMethodManagement;
+
+    /**
+     * @var EstimateAddressInterfaceFactory
+     */
+    private $estimateAddressFactory;
+
+    /**
+     * @param ScopeConfigInterface $scopeConfig
+     * @param CartRepositoryInterface $cartRepository
+     * @param ShippingMethodManagementInterface $shippingMethodManagement
+     * @param EstimateAddressInterfaceFactory $estimateAddressFactory
+     */
+    public function __construct(
+        ScopeConfigInterface $scopeConfig,
+        CartRepositoryInterface $cartRepository,
+        ShippingMethodManagementInterface $shippingMethodManagement,
+        EstimateAddressInterfaceFactory $estimateAddressFactory
+    ) {
+        $this->cartRepository = $cartRepository;
+        $this->scopeConfig = $scopeConfig;
+        $this->shippingMethodManagement = $shippingMethodManagement;
+        $this->estimateAddressFactory = $estimateAddressFactory;
+    }
+
+    /**
+     * Check if In-Store Pickup delivery method is applicable for a cart by cartId.
+     *
+     * @param int $cartId
+     *
+     * @return bool
+     */
+    public function execute(int $cartId): bool
+    {
+        try {
+            $cart = $this->cartRepository->get($cartId);
+            $address = $this->getEstimateAddress($cart);
+            // TODO: replace deprecated method usage
+            foreach ($this->shippingMethodManagement->estimateByAddress($cartId, $address) as $method) {
+                if ($method->getCarrierCode() == self::CARRIER_CODE) {
+                    return true;
+                }
+            }
+
+            return false;
+        } catch (NoSuchEntityException $e) {
+            return false;
```

Comment:
Could you leave only one 'return' statement ?

Reply:
Is it better to have an empty `catch` statement instead of 2 `return`s placed close to each other?
I am not sure about that

Reply:
Just have `$result = false` in the beginning, `$result = true` in the 'if' and `$result = false` in catch. Then - return.

Reply:
Genius